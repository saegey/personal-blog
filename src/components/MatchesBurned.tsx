import { useState } from 'react'
import { Box, Flex, Text, Button } from 'theme-ui'

import Modal from './Modal'

const MatchesHeading = () => {
  return (
    <Flex sx={{ height: ['35px', '35px', '35px'], marginTop: '20px' }}>
      <Box sx={{ width: '34%' }}>
        <Text as="p" variant="resultsItem">
          <strong>Average Power</strong>
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'center' }}>
        <Text as="p" variant="resultsItem">
          <strong>Total Joules</strong>
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'right' }}>
        <Text as="p" variant="resultsItem">
          <strong>Total time</strong>
        </Text>
      </Box>
    </Flex>
  )
}
type ItemProps = {
  index: number
  d: {
    averagePower: number
    totalJoules: number
    totalTime: number
    vals: number[]
    startTime: string
  }
}
const MatchesItem = ({ index, d }: ItemProps) => {
  return (
    <Flex key={`match${index}`} sx={{ height: ['35px', '35px', '35px'] }}>
      <Box sx={{ width: '34%' }}>
        <Text as="p" variant="resultsItem">
          {d.averagePower} watts
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'center' }}>
        <Text as="p" variant="resultsItem">
          {(d.totalJoules / 1000).toFixed(2)} kJ
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'right' }}>
        <Text as="p" variant="resultsItem">
          {d.totalTime} sec
        </Text>
      </Box>
    </Flex>
  )
}

type Props = {
  data: [
    {
      averagePower: number
      totalJoules: number
      totalTime: number
      vals: number[]
      startTime: string
    }
  ]
}

export const MatchesBurned = ({ data }: Props) => {
  const [isModalOpen, openModal] = useState(false)

  return (
    <>
      {isModalOpen && (
        <Modal modalOpen={openModal} headerText={"Matches Burned"}>
          <>
            <MatchesHeading />
            {data.map((d, index: number) => {
              return <MatchesItem d={d} index={index} key={index} />
            })}
          </>
        </Modal>
      )}
      <Box>
        <Flex>
          <Box sx={{ marginBottom: ['10px', '0px', '0px'] }}>
            <Text as="h2" variant="resultsHeading">
              Matches Burned
            </Text>
          </Box>
          <Box sx={{ marginLeft: 'auto' }}>
            <Button
              onClick={() => {
                openModal(true)
              }}
            >
              View All
            </Button>
          </Box>
        </Flex>
        <MatchesHeading />
        {data.slice(0, 5).map((d, index: number) => {
          return <MatchesItem d={d} index={index} />
        })}
      </Box>
    </>
  )
}
