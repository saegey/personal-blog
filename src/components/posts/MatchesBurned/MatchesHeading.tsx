import { Flex, Box, Text } from 'theme-ui'

const MatchesHeading = () => {
  return (
    <Flex sx={{ height: ['35px', '35px', '35px'], marginTop: '20px' }}>
      <Box sx={{ width: '34%' }}>
        <Text as="p" variant="resultsItem">
          <strong>Average Power</strong>
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'center' }}>
        <Text as="p" variant="resultsItem">
          <strong>Total Joules</strong>
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'right' }}>
        <Text as="p" variant="resultsItem">
          <strong>Total time</strong>
        </Text>
      </Box>
    </Flex>
  )
}

export default MatchesHeading
