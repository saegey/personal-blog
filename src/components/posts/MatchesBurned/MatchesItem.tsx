import { Flex, Box, Text } from 'theme-ui'

type ItemProps = {
  index: number
  d: {
    averagePower: number
    totalJoules: number
    totalTime: number
    vals: number[]
    startTime: string
  }
}

const MatchesItem = ({ index, d }: ItemProps) => {
  return (
    <Flex key={`match${index}`} sx={{ height: ['35px', '35px', '35px'] }}>
      <Box sx={{ width: '34%' }}>
        <Text as="p" variant="resultsItem">
          {d.averagePower} watts
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'center' }}>
        <Text as="p" variant="resultsItem">
          {(d.totalJoules / 1000).toFixed(2)} kJ
        </Text>
      </Box>
      <Box sx={{ width: '33%', textAlign: 'right' }}>
        <Text as="p" variant="resultsItem">
          {d.totalTime} sec
        </Text>
      </Box>
    </Flex>
  )
}

export default MatchesItem
