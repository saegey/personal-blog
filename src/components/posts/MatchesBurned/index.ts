import { MatchesBurned } from './MatchesBurned';

export default MatchesBurned
