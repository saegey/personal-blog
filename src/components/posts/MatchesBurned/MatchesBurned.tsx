import React from 'react'
import { Box, Flex, Text } from 'theme-ui'

import FullScreenIcon from '../../icons/FullScreenIcon'
import Modal from '../../Modal'
import MatchesHeading from './MatchesHeading'
import MatchesItem from './MatchesItem'

type Props = {
  data: Array<{
    averagePower: number
    totalJoules: number
    totalTime: number
    vals?: number[]
    startTime?: string
  }>
}

export const MatchesBurned = ({ data }: Props) => {
  const [isModalOpen, openModal] = React.useState(false)

  return (
    <>
      {isModalOpen && (
        <Modal
          modalOpen={openModal}
          headerText={'Matches Burned'}
          testId={'matches-modal'}
        >
          <>
            <MatchesHeading />
            {data.map((d, index: number) => {
              return <MatchesItem d={d} index={index} key={index} />
            })}
          </>
        </Modal>
      )}
      <Box
        sx={{
          maxWidth: '690px',
          margin: '60px auto',
          background: 'muted',
          padding: '30px',
          borderRadius: '5px',
        }}
      >
        <Flex>
          <Box sx={{ marginBottom: '20px' }}>
            <Text as="h2" variant="resultsHeading">
              Matches Burned
            </Text>
          </Box>
          <Box sx={{ marginLeft: 'auto' }}>
            <Box as="button"
              data-testid={'matches-burned-fullscreen'}
              sx={{
                width: '40px',
                right: '0',
                top: '0',
                zIndex: 0,
              }}
              onClick={() => {
                openModal(true)
              }}
            >
              <FullScreenIcon color="primary" />
            </Box>
          </Box>
        </Flex>
        <MatchesHeading />
        {data.slice(0, 5).map((d, index: number) => {
          return <MatchesItem d={d} key={`matches-${index}`} index={index} />
        })}
      </Box>
    </>
  )
}
