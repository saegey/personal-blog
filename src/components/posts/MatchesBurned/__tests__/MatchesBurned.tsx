/**
 * @jest-environment jsdom
 */
import renderer from 'react-test-renderer'
import { render, fireEvent, screen } from '@testing-library/react'
import { MatchesBurned } from '../MatchesBurned'
import React from 'react'

let data: Array<{
  averagePower: number
  totalJoules: number
  totalTime: number
  vals?: number[]
  startTime?: string
}>

describe('MatchesBurned', () => {
  beforeEach(() => {
    data = [
      {
        averagePower: 250,
        totalJoules: 1000,
        totalTime: 4,
      },
      {
        averagePower: 300,
        totalJoules: 3000,
        totalTime: 10,
      },
    ]
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('renders', () => {
    const json = renderer.create(<MatchesBurned data={data} />).toJSON()
    expect(json).toMatchSnapshot()
  })

  it('clicks full screen button to changes state to true', async () => {
    const setStateMock = jest.fn()
    let button: HTMLElement
    const useStateMock: any = (useState: any) => [useState, setStateMock]
    jest.spyOn(React, 'useState').mockImplementation(useStateMock)

    const { queryByTestId } = render(<MatchesBurned data={data} />)
    button = queryByTestId('matches-burned-fullscreen')!
    fireEvent.click(button)
    expect(setStateMock).toHaveBeenCalledWith(true)
  })

  it('clicks full screen button it renders the modal', async () => {
    render(<MatchesBurned data={data} />)
    const button = screen.queryByTestId('matches-burned-fullscreen')!
    fireEvent.click(button)
    const modal = screen.queryByTestId('matches-modal')
    expect(modal).toMatchSnapshot()
  })

  it('doesnt render modal on initial load', async () => {
    render(<MatchesBurned data={data} />)
    const modal = screen.queryByTestId('matches-modal')
    expect(modal).toBe(null)
  })
})
