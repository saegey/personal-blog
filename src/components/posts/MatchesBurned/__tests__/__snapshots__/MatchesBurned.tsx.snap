// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchesBurned clicks full screen button it renders the modal 1`] = `null`;

exports[`MatchesBurned renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 10px;
}

@media screen and (min-width: 40em) {
  .emotion-2 {
    margin-bottom: 0px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-2 {
    margin-bottom: 0px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  height: 32px;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 32px;
  height: 32px;
  right: 0;
  top: 0;
  z-index: 0;
  border: none;
  background-color: mutedAccent;
  border-radius: 5px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5:hover {
  background-color: mutedAccentMore;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  height: 100%;
  padding: 2px;
}

.emotion-6 .icon-solid {
  fill: undefined;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 35px;
  margin-top: 20px;
}

@media screen and (min-width: 40em) {
  .emotion-7 {
    height: 35px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-7 {
    height: 35px;
  }
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 34%;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 33%;
  text-align: center;
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 33%;
  text-align: right;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 35px;
}

@media screen and (min-width: 40em) {
  .emotion-14 {
    height: 35px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-14 {
    height: 35px;
  }
}

<figure
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <div
      className="emotion-2"
    >
      <h2
        className="emotion-0"
      >
        Matches Burned
      </h2>
    </div>
    <div
      className="emotion-4"
    >
      <button
        className="emotion-5"
        data-testid="matches-burned-fullscreen"
        onClick={[Function]}
      >
        <svg
          className="emotion-6"
          viewBox="0 0 14 14"
        >
          <g
            fill="none"
            fillRule="evenodd"
            id="Page-1"
            stroke="none"
            strokeWidth="1"
          >
            <g
              className="icon-solid"
              id="Core"
              transform="translate(-215.000000, -257.000000)"
            >
              <g
                id="fullscreen"
                transform="translate(215.000000, 257.000000)"
              >
                <path
                  d="M2,9 L0,9 L0,14 L5,14 L5,12 L2,12 L2,9 L2,9 Z M0,5 L2,5 L2,2 L5,2 L5,0 L0,0 L0,5 L0,5 Z M12,12 L9,12 L9,14 L14,14 L14,9 L12,9 L12,12 L12,12 Z M9,0 L9,2 L12,2 L12,5 L14,5 L14,0 L9,0 L9,0 Z"
                  id="Shape"
                />
              </g>
            </g>
          </g>
        </svg>
      </button>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-8"
    >
      <p
        className="emotion-0"
      >
        <strong>
          Average Power
        </strong>
      </p>
    </div>
    <div
      className="emotion-10"
    >
      <p
        className="emotion-0"
      >
        <strong>
          Total Joules
        </strong>
      </p>
    </div>
    <div
      className="emotion-12"
    >
      <p
        className="emotion-0"
      >
        <strong>
          Total time
        </strong>
      </p>
    </div>
  </div>
  <div
    className="emotion-14"
  >
    <div
      className="emotion-8"
    >
      <p
        className="emotion-0"
      >
        250
         watts
      </p>
    </div>
    <div
      className="emotion-10"
    >
      <p
        className="emotion-0"
      >
        1.00
         kJ
      </p>
    </div>
    <div
      className="emotion-12"
    >
      <p
        className="emotion-0"
      >
        4
         sec
      </p>
    </div>
  </div>
  <div
    className="emotion-14"
  >
    <div
      className="emotion-8"
    >
      <p
        className="emotion-0"
      >
        300
         watts
      </p>
    </div>
    <div
      className="emotion-10"
    >
      <p
        className="emotion-0"
      >
        3.00
         kJ
      </p>
    </div>
    <div
      className="emotion-12"
    >
      <p
        className="emotion-0"
      >
        10
         sec
      </p>
    </div>
  </div>
</figure>
`;
